src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -45,6 +45,14 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             // concats as well.  Walk to the top of that entire chain.
             var selectedExpression = token.Parent;
             var top = selectedExpression;
+
+            while (syntaxFacts.IsInterpolatedStringExpression(top) ||
+                syntaxFacts.IsInterpolation(top) ||
+                syntaxFacts.IsInterpolatedStringText(top))
+            {
+                top = top.Parent;
+            }
+
             while (IsStringConcat(syntaxFacts, top.Parent, semanticModel, cancellationToken))
```

Comment:
at 72, I am not sure why this refactoring choose to not offer refactoring? this is refactoring so, this seems arbitary reason not to offer it. we should just offer it and it is up to the user whether they will use it or not.

// If the entire expression is just concatenated strings, then don't offer to
// make an interpolated string.  The user likely manually split this for
// readability.